### ISSUE TITLE
RxJava2 support

### ISSUE THREAD

**Issue by adauvalter** [greenrobot/ObjectBox#1]
E.g. subscription to entity changes.

**Comment by greenrobot** [comment]
Brainstorming:
The basis will be non-RxJava notifications with listeners #15.

**When should notification be emitted?** Right after a put() call or once the transaction completed?
--> After put(), changes would only be visible to the current transaction. Also the transaction may be roll backed. So it seems to make much more sense to fire notifications once a TX is committed.

As a first step we could just notify that changes occurred in entity type X without specifying which entities. Or a list of entities types (X,Y, and Z).

Later (if helpful), we could offer change sets (putted and removed entity IDs).

**Comment by FabianTerhorst** [comment]
@greenrobot you have any release date for change listener support? This is probably the most used missing feature right now.

**Comment by greenrobot** [comment]
@FabianTerhorst Not sure if that is the most wanted feature, but current plans put #15 right after #18 and #18.

**Comment by sokarcreative** [comment]
I totally agree with FabienTerhost, a change listener feature would be appreciated

**Comment by greenrobot** [comment]
FYI: There's no RxJava extension yet, but please check Version 0.9.8-RC2 (or later). Look for BoxStore.subscribe() and Query.subscribe(). Details follow.

**Comment by greenrobot** [comment]
Here is a first version: https://github.com/greenrobot/ObjectBoxRxJava
Would be happy about feedback.

**Comment by greenrobot** [comment]
Related: http://greenrobot.org/objectbox/documentation/data-observers-reactive-extensions/

**Comment by greenrobot** [comment]
I'm sure this can be improved, but there is initial support released in 0.9.8. If you have feedback, please create a new issue.
